### ISSUE TITLE
Error 403 Forbidden - CLIENT_OR_AGENT_NOT_AUTHORISED

### ISSUE THREAD

**Issue by nicol1993** [hmrc/vat-api#355]
Hi All,

I'm getting the following message response when I try to Retrieve VAT obligations:

           **CLIENT_OR_AGENT_NOT_AUTHORISED**

I'm using a test user created through the API rather than the GUI as I saw in previous threads that it can be a bit tempermental through the GUI. 

Here is the request URL using GET: 

**https://test-api.service.hmrc.gov.uk/organisations/vat/123456789/obligations?from=2018-10-01&to=2018-10-31**
 

Headers:
Content-Type   application/json
Accept              application/vnd.hmrc.1.0+json
Authorization    Bearer [MY_GENERATED_ACCESS_TOKEN]

Any help or suggestions would be appreciated.

Thanks

Nic

**Comment by GrahamHDawson** [comment]
Don't know if the request URL you quote above is 'literal' - but if it **is** then I doubt the VRN '123456789' is the VRN of a created test user.
Having said that I wouldn't expect the error to be NOT_AUTHORISED - that points to the access token you're using being invalid.

How did you get that?
Did you successfully get an authorisation code and exchange that for an access token and refresh toke?

Graham

**Comment by nicol1993** [comment]
Hi Graham,

Thanks for your reply.
Ah okay, i just copied the one from the API documentation. I'm using 666429721 now.

Yeah I managed to get an authorisation token then an access token. Here's the output when I exchange the tokens.

200 OK
{"access_token":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","refresh_token":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","expires_in":14400,"scope":"hello read:vat write:vat","token_type":"bearer"}

I then copy that token for use in the Authorization header. I'll also mention that the access token works for creating a test user through the API as that requires an OAuth Bearer token.

Thanks

Nic

**Comment by GrahamHDawson** [comment]
Hi Nic,

Couple of points...
1 - "scope": "hello read:vat write:vat" - that's wrong you can't use the hello end-point for the MTD VAT API, have you registered your application and subscribed to MTD VAT?
2 - 'the access token works for creating a test user' - you should use your ServerToken for that not an access token. If you use an access token then you need to create a test user to get an access token to create a test user....
Whereas if you use the ServerToken that you're give when you register for the API you can create a test user 'from scratch' :-)

Graham

**Comment by GrahamHDawson** [comment]
...when you get your authorisation code the URL should have  this..
&scope=read:vat write:vat&state= etc etc ie no 'hello' in there.

You **may** need to add a + character in there ie &scope=read:vat+write:vat&state etc - it all depends how you're doing it.

To be honest I'm very surprised that scope=hello read:vat write:vat was accepted and didn't result in an 'Invalid Scope' error.

Graham

**Comment by GrahamHDawson** [comment]
Hi Nic,

If you log into your developer account at HMRC and use the 'Manage API Subscriptions' what do you see?

You should have 1 subscription against the MTD VAT API see below...
![mtdvatsubscription](https://user-images.githubusercontent.com/42624873/49219898-93011600-f3cc-11e8-8160-cc720d39682c.PNG)

You can't use the VAT API against a Hello World subscription.

Graham

**Comment by nicol1993** [comment]
Hi Graham,

I have a subscription to the VAT (MTD) API and I did indeed have a subscription to the Hello World API which I have now disabled.

![image](https://user-images.githubusercontent.com/28361831/49220853-3b17de80-f3cf-11e8-9757-fab06f2d5df6.png)

![image](https://user-images.githubusercontent.com/28361831/49220919-6ac6e680-f3cf-11e8-9089-49f5d9bc5fcc.png)

I recreated a test user with server token and tried to use the new access token to call the obligations but no joy unfortunately. Still getting a 403.
